firestore-bigquery-export/firestore-bigquery-change-tracker/src/logger.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { logger as funcsLogger } from "firebase-functions";
+
+export enum LogLevel {
+  DEBUG = "debug", // Will log everything
+  INFO = "info", // Will log info, warnings, and errors
+  WARN = "warn", // Will log warnings and errors
+  ERROR = "error", // Will log errors only
+  SILENT = "silent", // Won't log anything
+}
+
+const levels = {
+  debug: 0,
+  info: 1,
+  warn: 2,
+  error: 3,
+  silent: 4,
+};
+
+export class Logger {
+  private logLevel: number;
+
+  constructor(logLevel: LogLevel | string | number = LogLevel.INFO) {
+    this.setLogLevel(logLevel);
+  }
+
+  setLogLevel(logLevel: LogLevel | string | number): void {
+    if (typeof logLevel === "string") {
+      this.logLevel = levels[logLevel];
```

Comment:
```suggestion
      this.logLevel = levels[logLevel as keyof typeof levels] ?? levels.info;
```

or `logLevel as string` maybe

the main thing is the fallback `?? levels.info`  or something equivalent, just in case a bad string comes through and we'd have `undefined`